sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/NumberConverterSuite.scala

```diff
@@ -55,6 +55,12 @@ class NumberConverterSuite extends SparkFunSuite {
     checkConv("-10", 11, 7, "45012021522523134134555")
   }
 
+  test("SPARK-44973: conv must allocate enough space for all digits plus negative sign") {
+    checkConv(s"${Long.MinValue}", 10, -2, "-1" + "0" * 63)
+    checkConv("8" + "0" * 15, 16, -2, "-1" + "0" * 63)
```

Comment:
String construction is a little convoluted, we could use `BigInt` to make it more obvious, e.g.:

```Scala
BigInt(Int.MinValue).toString(2)
```

Reply:
Done (although IMO this is less obvious in some sense)

Reply:
It was not meant as a blocking comment either way